pacman/operations/placer_algorithms/one_to_one_placer.py

```diff
@@ -52,166 +115,97 @@ def _do_allocation(
             machine, self._generate_radial_chips(machine))
         all_vertices_placed = set()
 
-        # iterate over vertices
-        for vertex_list in vertices:
+        # Find vertices with harder constraints
+        constrained = list()
+        unconstrained = list()
+        for vertex in machine_graph.vertices:
+            if locate_constraints_of_type(
+                    vertex.constraints, ChipAndCoreConstraint):
+                constrained.append(vertex)
+            else:
+                unconstrained.append(vertex)
+
+        # Place vertices with hard constraints
+        for vertex in constrained:
+            self._allocate_individual(
+                vertex, placements, resource_tracker, same_chip_vertex_groups,
+                all_vertices_placed, progress)
+
+        # iterate over the remaining unconstrained (or less constrained)
+        # vertices
+        for vertex in unconstrained:
+
+            # If the vertex has been placed, skip it
+            if vertex in all_vertices_placed:
+                continue
+
+            # Find vertices that are one-to-one connected to this one
+            one_to_one_vertices = one_to_one_groups[vertex]
```

Comment:
mmm, this makes me think its better merged, as its considered unconstrained or constrained above... based off placement (and in future sdram links) (Note for myself that branch will need a big conflict merge).

Reply:
Note SDRAM links already exist here (but don't have any resources associated with the edge).

Reply:
i noticed after typing.